FEATURE: Add workspace Sync button

pKallert opened the pull request:
This will add a rebase button to the UI for rebasing the personal workspace of a user

**What I did**
I added a button to the Neos UI that shows up if the personal workspace of a user is outdated or in error state.

**How I did it**
Neos UI when personal workspace is outdated: 
![Bildschirmfoto 2023-11-24 um 12 04 32](https://github.com/neos/neos-ui/assets/91674611/14a943b7-97cf-4554-b404-2353a8f26848)

Clicking on the button shows popup: 
![Bildschirmfoto 2023-11-24 um 11 57 55](https://github.com/neos/neos-ui/assets/91674611/b2772c51-4115-41b9-81ce-a7bf28c251aa)

Clicking on "Synchronize now" will rebase the workspace

Neos UI when personal workspace is outdated and in error mode: 
![Bildschirmfoto 2023-11-24 um 12 03 03](https://github.com/neos/neos-ui/assets/91674611/7f80e91c-e704-465c-917f-da735ea5316a)

Clicking on the button shows popup: 
![Bildschirmfoto 2023-11-24 um 12 03 14](https://github.com/neos/neos-ui/assets/91674611/0cc0c141-1ed5-4cfd-b826-1a2e718b4692)

Related: https://github.com/neos/neos-ui/issues/3803

pKallert committed:
Feature: Add basic workspace sync button

pKallert committed:
Feature: Add popup for workspace rebase

pKallert committed:
Fix: code style

pKallert committed:
Feature: adjust icon styling

pKallert committed:
Feature: Introduce Enum

pKallert committed:
Feature: Update comments

pKallert committed:
Feature: Add loading icon and status

markusguenther committed:
TASK: Use font-awesome layer icons

Replaces custom SVG icons with font=awesome icons.

markusguenther commented:
I replaced the custom Icons with default `fontawesome` icons. As the inner circle is too small for additional signs, I added a badge as a layer. The result looks now like that:

**Up-to-date:**
<img width="749" alt="Screenshot 2023-12-11 at 13 28 27" src="https://github.com/neos/neos-ui/assets/1014126/b08c7d37-57bb-4f68-83bd-9c64603b8147">

**Outdated but can be rebased:**
<img width="781" alt="Screenshot 2023-12-11 at 13 28 06" src="https://github.com/neos/neos-ui/assets/1014126/de452e45-302f-4ed8-8f6c-b88ba3d5c91c">

**Conflict:**
<img width="792" alt="Screenshot 2023-12-11 at 13 17 43" src="https://github.com/neos/neos-ui/assets/1014126/ddb51fc1-4d2b-4398-b7ea-9a3bdf2fae5c">

Also, the dialog looks a bit different now.

**Outdated but can be rebased:**
<img width="677" alt="Screenshot 2023-12-11 at 15 54 05" src="https://github.com/neos/neos-ui/assets/1014126/992ea58c-f517-4822-9417-2a5e6a715b11">

**Conflict:**
<img width="685" alt="Screenshot 2023-12-11 at 15 54 34" src="https://github.com/neos/neos-ui/assets/1014126/181bd831-199f-4c8d-941c-83d21465f036">

markusguenther commented:
I also taking care of the end to end tests.

markusguenther committed:
TASK: Prevent editing with conflicting workspace

When the workspace has conflicts that cannot be resolved, it does not make sense to be able to edit the content. Therefore, we mark the workspace as read only in this case.

markusguenther commented:
Is it OK that I additionally mark the workspace as read only, so that users cannot edit content when the workspace is falling in a conflicting state? It makes no sense to be able to edit content when it cannot be used anyway.

mhsdesign commented on packages/neos-ui-redux-store/src/CR/Workspaces/index.ts:
```diff
@@ -75,6 +78,11 @@ const confirmDiscard = () => createAction(actionTypes.DISCARD_CONFIRMED);
+/**
+ * Rebase the user workspace
+ */
+const rebaseWorkspace = (name: string) => createAction(actionTypes.REBASE_WORKSPACE, name);
```
is there no reducer needed for REBASE_WORKSPACE? i guess because its handled via saga?

mhsdesign reviewed (commented):
The code looks good so far and i also like the icon and gui :heart:

But i noticed some things while testing:

- if changes were made to the currently viewed page and a rebase is excecuted the document must be reloaded (and maybe also the page tree generally?)
- when rebasing there are two spinners which seems odd?
<img width="316" alt="image" src="https://github.com/neos/neos-ui/assets/85400359/ba45901b-261c-4f16-ba12-7be8454bc558">
- one can only rebase the user workspace and detect conflicts there. But if the base workspace is a custom workspace and not live and it is outdated or in conflict the ui doesnt tell

bwaidelich commented on Classes/Controller/BackendServiceController.php:
```diff
@@ -618,4 +619,34 @@ public function generateUriPathSegmentAction(string $contextNode, string $text):
+        }
+
+        $success = new Success();
+        $success->setMessage(sprintf('Successfully synced User workspace'));
```
Is that something that needs to be localized?

markusguenther replied:
Indeed will take care later!

mhsdesign replied:
It might be that those messages are not a flashmessage but just show up in the console as debug info in which case translation is irrelevant

pKallert replied:
We should probably then also localize all of the other $success->setMessage calls in the Controller to be consistent

bwaidelich reviewed (commented):
I can't really judge on the ReactJS parts though and I didn't get around testing this yet but everything that I understand looks really neat and great! :)

markusguenther commented:
@mhsdesign the spinner is a known issue but as it reacts on the same states we would need a new state then for the rebase in process 🤷‍♂️

markusguenther committed:
FEATURE: Localize rebase response

markusguenther committed:
TASK: Add on remote state for synchronization

markusguenther commented:
@bwaidelich @mhsdesign Hope that I could fulfil your wishes :), we have now translator labels and an extra remote state and therefore also a reducer, so now the indicator is just for the rebase.

dlubitz reviewed (approved):
Looks good to me by reading! Thank you @pKallert for taking care!
